_posts/2019-12-10-bert-model.md

```diff
@@ -0,0 +1,410 @@
+(left-to-right or right-to-left), the Transformer encoder reads the entire
+sequence of words at once. Therefore it is considered bidirectional, though
+it would be more accurate to say that it’s non-directional. This characteristic
+allows the model to learn the context of a word based on all of its surroundings
+(left and right of the word).
+
+In BERT, we don't represent each word as a vector (as we've been doing since
+years in NLP) instead we represent each 'subword token' as a vector. That is,
+whenever the model finds an unknown word it tries to split the word into
+sub-pieces like the word root, prefixes, etc  and use all the representational
+vectors of each sub-pieces together to represent the unfamiliar word.
+
+![bert-work-1](../assets/images/bert-work-1.png)
+
+Finally, we take the output of the last encder block as our final representation.
+We can just feed the representation of our old NLP model to BERT and get a much
+better performance. This means that the BERT isn't as complicated as some of the
+other NLP models. However, the BERT model is VERY large. As an instance,
+for the Aamzon Review dataset I wasn't able to train 10 reviews on my 6 GB GPU.
+Which I think is insane! I'll get into more details in the upcoming sections.
+
+Well, that's for the general introduction to the BERT model. Its actually not
+complete, but I will mention those things over my own task to avoid redundancy
+in information.
+
+## What was my task and how I did it?
+
+My task was to train and test
+[Amazon Review dataset](https://www.kaggle.com/bittlingmayer/amazonreviews)
+using BERT model present in
+[Transformers.jl](https://github.com/chengchingwen/Transformers.jl) package.
+
+One of the nicest things of this task was that I was not aware of any such
+model. One of the key reasons of me participating and contributing to Julia
+was I wanted to get into Machine Learning and I wanted to learn a new language.
+Being introduced to such algorithms is I think one of the best things I could
+ever get. This task was also tagged as 'Advanced' and while I've completed it,
+I clearly understand why.
+
+In order to complete the task, I need to have a complete basic understanding of
+the model. It was a tough job, and while reading various blogs, papers, articles
+it was hard to process all that information. There are I think a lot of factors
+which I think only experienced ML people would get at first instance. I had to
+Google almost every new term I had discovered while reading. So, I spent time
+reading and understanding and after I had a basic idea, I was ready to look at
+the example code.
+
+The Transformers.jl repository already has few examples on BERT. Here we are
+working on Sentiment Analysis and hence we need to construct a classifier. We
+will now see the code and the output for the model I constructed. Its a bit
+long, and I won't be able to explain each and every section of the code. But,
+I'll cover up the important ones.
+
```

Comment:
Say a little about what you are classifiying. What are you trying to predict? What are the lables?

Reply:
Sure, I'll modify this